crates/sovereign-sdk/rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -242,6 +242,29 @@ pub struct BatchProofCircuitInputV2<'txs, StateRoot, Witness, Da: DaSpec, Tx: Cl
     pub sequencer_commitments_range: (u32, u32),
 }
 
+/// Grouping circuit input to determine which to use based on current fork.
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
+// Prevent serde from generating spurious trait bounds. The correct serde bounds are already enforced by the
+// StateTransitionFunction, DA, and Zkvm traits.
+#[serde(
+    bound = "StateRoot: Serialize + DeserializeOwned, Witness: Serialize + DeserializeOwned, Tx: Serialize + DeserializeOwned"
+)]
+pub enum BatchProofCircuitInput<'txs, StateRoot, Witness, Da: DaSpec, Tx: Clone> {
+    /// V1 or genesis
+    V1(BatchProofCircuitInputV1<'txs, StateRoot, Witness, Da, Tx>),
+    /// V2
+    V2(BatchProofCircuitInputV2<'txs, StateRoot, Witness, Da, Tx>),
+}
+
+/// Grouping circuit output to determine which to use based on current fork.
+#[derive(Clone, Debug, Serialize, Deserialize, BorshSerialize, BorshDeserialize, PartialEq, Eq)]
+pub enum BatchProofCircuitOutput<Da: DaSpec, StateRoot> {
```

Comment:
one problem I know you'll have is that you are adding a byte for the outter enum now, you wont be able to parse testnet zk proofs

Reply:
I mean the ones we already sent

Reply:
maybe since we are only in testnet, we can forget about backwards compatability for these kind of things 🤔 

but it'll be hard to initialize light client proof circuit nevertheless

Reply:
I stopped yesterday because of this exact reason. Problem is, unless we are able to know what we're parsing, v1/v2, we won't have access to the last l2 height from which we can figure out the fork and therefore the output type. I have to find an alternative.

Reply:
one solution is to try parsing V2, if fails, go with v1

but that's not really a always working solution 😓

Reply:
Yea... and in this case we have to change the way `zkvm.read_from_host();` behaves since we consume stdin bytes once and try to parse them.